Sources/_MatchingEngine/Regex/Parse/LexicalAnalysis.swift

```diff
@@ -704,7 +861,8 @@ extension Source {
   ///     ExpGroupStart -> '(_:'
   ///
   mutating func lexAtom(
-    isInCustomCharacterClass customCC: Bool
+    isInCustomCharacterClass customCC: Bool,
+    numberOfPriorGroups: Int
```

Comment:
Naming convention nit: `...Count` is much more commonly used than `numberOf...`. Suggest `priorGroupCount:`.